samples/performance/constant_data/constant_data_tutorial.md

```diff
@@ -0,0 +1,405 @@
+
+*You can read more about interface blocks [here](https://www.khronos.org/opengl/wiki/Interface_Block_(GLSL)).*
+
+### **Vulkan API**
+
+We've covered how constant data is implemented in the shader, however to push the data from the application to the shader we need to use Vulkan.
+
+We do this with the use of `VkBuffer`s, which is Vulkan's implementation of buffer memory.
+
+Buffers in Vulkan are just chunks of memory used for storing data, which can be read by the GPU.
+
+They need to be created and have their memory *manually* allocated, and then we can copy our constant data into the allocated memory. This data can then be plugged into the draw calls, so that it can finally be used in our shader computations. 
+
+**Note:** *The library [Vulkan Memory Allocator (VMA)](https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator) is extremely good for handling a lot of the common pitfalls that come with handling your Vulkan memory without removing the control that you would otherwise have with native Vulkan.*
+
+The following links are useful for learning how to create a Vulkan buffer in your application:
+
+* https://vulkan-tutorial.com/Vertex_buffers/Vertex_buffer_creation
+* https://vulkan-tutorial.com/Vertex_buffers/Staging_buffer#page_Abstracting-buffer-creation
+* https://vulkan-tutorial.com/Uniform_buffers/Descriptor_layout_and_buffer#page_Uniform-buffer
+
+#### *The Methods*
+
+Generally, there are two ways to push your constant data. It is either with **push constants** or **descriptor sets**. 
+
+However the Vulkan API gives a lot of flexibility about how to handle **descriptor sets**, offering many different types and different ways to bind and use them (especially when we factor in extensions). This can puzzle developers about which is best, and for which scenarios. This tutorial aims to ease some of the confusion and uncertainty around this subject. 
+
+When we break this down, we have the following methods:
+
+* Push Constants
+* Descriptor Sets
+* Dynamic Descriptor Sets
+* Update-after-bind Descriptor Sets
+* Buffer array with dynamic indexing
+* [Inline uniform buffer objects](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VK_EXT_inline_uniform_block.html) (click to read more)
+* [Push descriptors](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VK_KHR_push_descriptor.html) (click to read more)
+
+**Inline uniform buffer objects** and **push descriptors** are not covered by this tutorial, please use the links above to learn more about them.
+
+## **Sample Overview**
+
+### **Introduction**
+
+The sample uses a mesh heavy scene which has 1856 meshes (475 KB of mesh data). This is to demonstrate a use case where many different calls to pushing constant data will occur during a single frame. 
```

Comment:
I think we need to point out that having lots of constant data for each mesh like this is not recommended practice but is being used to artificially exaggerate the performance differences in this example.